Eliminar Registros/Imagen

itzel-aguirre opened the pull request:
## ¿Qué hice?

- Al seleccionar el botón de eliminar, borra los registros correspondientes al Espectáculo:
 Reservaciones
 Fechas
 Generos
 Espectáculos  
 Imágenes
- Carga los espectáculos que aun están disponibles
- Envía mensaje de notificación 

## Ejemplo

itzel-aguirre committed:
Eliminar imagenes de la carpeta

itzel-aguirre committed:
Arreglando conflictos del archivo index.php

Merge remote-tracking branch 'upstream/master' into feature/deleteImage

amilkhael commented on model/ShowBO.php:
```diff
@@ -166,4 +166,8 @@ public function ListShedules($id_show){
 
     return $schedules;
   }
+
```
Esttá vaciío jejeje creo te falto hacer commit y luego push

itzel-aguirre committed:
Eliminar registros e imagenes de la pantalla del Admin

itzel-aguirre committed:
Arreglo de conflicto en genre.js

Merge remote-tracking branch 'upstream/master' into feature/deleteImage

amilkhael commented on controller/controller-deleteImage.php:
```diff
@@ -0,0 +1,22 @@
+catch(Exception $e){
+  header('HTTP/1.1 420 Method Failure');
+  header('Content-Type: application/json; charset=UTF-8');
+  die(json_encode (array('error'=>'Error al crear un nuevo show')));
```
Si la acción es delete porque el error dice error al crear un nuevo show?

amilkhael commented on controller/controller-deleteShow.php:
```diff
@@ -0,0 +1,22 @@
+else{
+  header('HTTP/1.1 420 Method Failure');
+  header('Content-Type: application/json; charset=UTF-8');
+  die(json_encode (array('error'=>'Incorrecto')));
```
Se más específica en el error

amilkhael commented on js/shows.js:
```diff
@@ -142,7 +142,9 @@ jQuery(function($) {
 jQuery(function($) {
   $("#table-shows tbody").on("click", "button.delete-show", function() {
-    alert($(this).attr("show-id"));
+    //alert($(this).attr("show-id"));
```
Quita el código comentdao

amilkhael commented on js/shows.js:
```diff
@@ -221,4 +223,47 @@ function resetForm(){
+  };
+  $.ajax({
+    type: "POST",
+    url: "controller/controller-deleteImage.php",
```
Yo creoq eu no es necesario el contraldor de delete image, al final puedes mandar a llamar el mñetodo cuando terminas d eborrar los registros y no es necesario este contrlador

amilkhael commented on js/shows.js:
```diff
@@ -221,4 +223,47 @@ function resetForm(){
+        $("#list-show").show();
+      },
+      error: function(errMsg) {
+        console.error(errMsg);
```
En que momento se le notifica al usuario del errore creo te hace falta probar que muestre la notificacion de error

amilkhael commented on js/shows.js:
```diff
@@ -221,4 +223,47 @@ function resetForm(){
+      dataType: "json",
+      success: function(showData) {
+        notifications ("Espectaculo eliminado exitosamente.", 'success')
+        $("#add-newShow").hide();
```
esto no es necesario no se mostraba el formulario del nuevo show

amilkhael commented on js/shows.js:
```diff
@@ -221,4 +223,47 @@ function resetForm(){
+        notifications ("Espectaculo eliminado exitosamente.", 'success')
+        $("#add-newShow").hide();
+        getShowList()
+        $("#list-show").show();
```
esto tampoco porque ya estás en esa vista

amilkhael commented on js/shows.js:
```diff
@@ -142,7 +142,9 @@ jQuery(function($) {
-    alert($(this).attr("show-id"));
+    //alert($(this).attr("show-id"));
+    const idShow = $(this).attr("show-id");
+    deleteImage(idShow);
```
Aquí recordemos hacer un Modal para preguntarle de verdad deseas eliminarlo?

itzel-aguirre replied:
El mensaje sería algo así?

El espectáculo será eliminado. ¿Desea continuar?
y dos opciones Si   o CANCELAR

itzel-aguirre committed:
Correcciones
- Eliminar regitros y archivos
- Limpiar formulario al guardar/cancelar un nuevo espectaculo
- Agregue Group by en consulta de espectaculos

itzel-aguirre commented:
Cambios realizados 
Solo tengo duda del que te pregunte

amilkhael commented on js/shows.js:
```diff
@@ -142,7 +143,9 @@ jQuery(function($) {
   $("#table-shows tbody").on("click", "button.delete-show", function() {
-    alert($(this).attr("show-id"));
+    const idShow = $(this).attr("show-id");
+    //deleteImage(idShow);
```
Trata de no dejar código comentado

amilkhael commented on js/shows.js:
```diff
@@ -221,4 +225,27 @@ function resetForm(){
+      dataType: "json",
+      success: function(showData) {
+        console.log(showData);
+        notifications ("Espectaculo eliminado exitosamente.", 'success')
```
También para producción quita los consolé. Log
